From: "Rémi Bernon" <rbernon@codeweavers.com>
Subject: [PATCH 3/3] winepulse.drv: Keep buffer size separate from allocated size.
Message-Id: <20211026153716.4093289-3-rbernon@codeweavers.com>
Date: Tue, 26 Oct 2021 17:37:16 +0200
In-Reply-To: <20211026153716.4093289-1-rbernon@codeweavers.com>
References: <20211026153716.4093289-1-rbernon@codeweavers.com>

So that it rounds to integral audio frame count.

Signed-off-by: Rémi Bernon <rbernon@codeweavers.com>
---

Not entirely sure if this is useful, but it's done differently for
render or capture allocation. This does it the same way as capture.

 dlls/winepulse.drv/pulse.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dlls/winepulse.drv/pulse.c b/dlls/winepulse.drv/pulse.c
index 10260dea106..f5d3a1ddca4 100644
--- a/dlls/winepulse.drv/pulse.c
+++ b/dlls/winepulse.drv/pulse.c
@@ -1065,7 +1065,7 @@ static NTSTATUS pulse_create_stream(void *args)
             stream->alloc_size = stream->real_bufsize_bytes =
                 stream->bufsize_frames * 2 * pa_frame_size(&stream->ss);
             if (NtAllocateVirtualMemory(GetCurrentProcess(), (void **)&stream->local_buffer,
-                                        0, &stream->real_bufsize_bytes, MEM_COMMIT, PAGE_READWRITE))
+                                        0, &stream->alloc_size, MEM_COMMIT, PAGE_READWRITE))
                 hr = E_OUTOFMEMORY;
         } else {
             UINT32 i, capture_packets;

-- 
2.33.0

